src/passes/Inlining.cpp

```diff
@@ -292,9 +292,16 @@ doInlining(Module* module, Function* into, const InliningAction& action) {
   // Zero out the vars (as we may be in a loop, and may depend on their
   // zero-init value
   for (Index i = 0; i < from->vars.size(); i++) {
+    auto type = from->vars[i];
+    if (type.isRef() && !type.isNullable()) {
+      // Non-nullable locals do not need to be zeroed out. They have no zero
+      // value, and by definition should not be used before being read from, so
```

Comment:
```suggestion
      // value, and by definition should not be used before being written to, so
```
Is this what you meant?

Reply:
Yes, good catch.